okhttp/src/main/kotlin/okhttp3/RequestBody.kt

```diff
@@ -133,6 +135,23 @@ abstract class RequestBody {
       }
     }
 
+    /** Returns a new request body that transmits this. */
+    @JvmStatic
+    @JvmName("create")
+    fun FileDescriptor.toRequestBody(contentType: MediaType? = null): RequestBody {
+      return object : RequestBody() {
+        override fun contentType() = contentType
+
+        override fun isOneShot(): Boolean = true
```

Comment:
Is this true? We can't read it twice?

Reply:
In theory you could reset the FileInputStream via a FileChannel, but it would seem quite a stretch.  Consider other frequent FileDescriptors like Pipes and STDIN.  It doesn't make sense in those cases.

At the point you want to reset it you should probably implement your own.

```
    /**
     * A handle to the standard input stream. Usually, this file
     * descriptor is not used directly, but rather via the input stream
     * known as <code>System.in</code>.
     *
     * @see     java.lang.System#in
     */
    public static final FileDescriptor in = new FileDescriptor(0);
```

Reply:
There is a test that confirms it fails, which doesn't rely on isOneShot.